build/gen_stub.php

```diff
@@ -1486,17 +1489,300 @@ private function getFlagsAsArginfoString(): string
      * @throws Exception
      */
     public function getMethodSynopsisDocument(array $funcMap, array $aliasMap): ?string {
-
-        $doc = new DOMDocument();
+        $doc = new DOMDocument("1.0", "utf-8");
         $doc->formatOutput = true;
+
+        $refentry = $doc->createElement('refentry');
+        $doc->appendChild($refentry);
+
+        if ($this->isMethod()) {
+            assert($this->name instanceof MethodName);
+            $id = $doc->createAttribute("xml:id");
+            $id->value = addslashes(strtolower($this->name->className->__toString())) . '.' . strtolower($this->name->methodName);
+            $refentry->appendChild($id);
+        } else {
+            // TODO Functions
+        }
+        $refentry->setAttribute("xmlns", "http://docbook.org/ns/docbook");
+        $refentry->appendChild(new DOMText("\n "));
+
+        /* Creation of <refnamediv> */
+        $refnamediv = $doc->createElement('refnamediv');
+        $refnamediv->appendChild(new DOMText("\n  "));
+        if ($this->isMethod()) {
+            assert($this->name instanceof MethodName);
+            $refname = $doc->createElement('refname', $this->name->className->__toString() . '::' . $this->name->methodName);
```

Comment:
This should be equivalent:
```suggestion
            $refname = $doc->createElement('refname', $this->name->__toString());
```

Reply:
And then you don't need the assert above